common/client-libs/gateway-client/src/client.rs

```diff
@@ -379,21 +379,26 @@ impl GatewayClient {
                 ws_stream,
                 self.local_identity.as_ref(),
                 self.gateway_identity,
-                self.coconut_credential.clone(),
             )
             .await
             .map_err(GatewayClientError::RegistrationFailure),
             _ => unreachable!(),
         }?;
-
-        self.authenticated = true;
-        Ok(shared_key)
+        self.authenticated = match self.read_control_response().await? {
+            ServerResponse::Register { status } => Ok(status),
+            ServerResponse::Error { message } => Err(GatewayClientError::GatewayError(message)),
+            _ => unreachable!(),
```

Comment:
Maybe for clarity error out with what came out of the `read_control_response`, since we're returning an enum at some point in the future someone might extend the function to also return some other enum member, and at that time we blow up

Reply:
I used `unreachable!` as it is already part of the flow on the authentication path, but we can switch to error returns as well. The end result should be the same (the client blows up) because of some protocol mismatch.